### PR TITLE
Resolve symlink of the `matlab` executable when finding install path

### PR DESCRIPTION
**Pull request by MoetaYuko**
Fixing https://github.com/mathworks/MATLAB-extension-for-vscode/issues/42

### CODE REVIEW

File: src/providers/linting/LintingSupportProvider.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { MatlabConnection } from '../../lifecycle/MatlabCommunicationManager'
 import MatlabLifecycleManager from '../../lifecycle/MatlabLifecycleManager'
 import Logger from '../../logging/Logger'
 import * as path from 'path'
+import fs = require('fs')
```
Comment (dklilley) [inline src/providers/linting/LintingSupportProvider.ts:12]: Please use `import * as fs from 'fs'` syntax instead

File: src/providers/linting/LintingSupportProvider.ts [hunk 2]
```diff
@@ -339,7 +340,11 @@ class LintingSupportProvider {
             try {
                 const resolvedPath = await which('matlab')
                 if (resolvedPath !== '') {
-                    binPath = path.dirname(resolvedPath)
+                    await fs.realpath(resolvedPath, (err, realPath) => {
+                        if (!err) {
+                            binPath = path.dirname(realPath)
+                        }
```
Comment (dklilley) [inline src/providers/linting/LintingSupportProvider.ts:346]: If there is an error, it would be good to log it using `Logger`

**Review by dklilley — CHANGES_REQUESTED** [review]
Thanks for looking into this!

**Comment by nothans** [comment]
Can you contact me to sign a CLA? Thank you for the PR!

**Comment by nothans** [comment]
The CLA has been signed. Please evaluate the PR. Thank you for the contribution!

**Comment by dklilley** [comment]
Hi @MoetaYuko, I am trying this out but I think this is not behaving as desired. When I try this, I am seeing that line 356 is executed before the `fs.realpath` callback is executed, resulting in a null return every time.

Is this what you are seeing as well?

**Comment by MoetaYuko** [comment]
> Hi @MoetaYuko, I am trying this out but I think this is not behaving as desired. When I try this, I am seeing that line 356 is executed before the `fs.realpath` callback is executed, resulting in a null return every time.
> 
> Is this what you are seeing as well?

Nope, it simply works as expected on my side.
* I didn't manually set matlabInstallPath so that the "else" block on line 338 is entered, then the "try" block.
* I have a symlink `/usr/bin/matlab -> /opt/MATLAB/R2023a/bin/matlab` where `/usr/bin` is in $PATH obviously. Before this PR `binPath` was assigned `/usr/bin`, but now it's `/opt/MATLAB/R2023a/bin` which is expected.

**Comment by dklilley** [comment]
I think this may be working for you somewhat by luck. Since you are using the non-promise version of `fs.realpath`, I don't believe the `await` has any effect. If it resolves fast enough, the callback function may be executed before the `try` is exited, however this is not guaranteed.

**Comment by MoetaYuko** [comment]
> I think this may be working for you somewhat by luck. Since you are using the non-promise version of `fs.realpath`, I don't believe the `await` has any effect. If it resolves fast enough, the callback function may be executed before the `try` is exited, however this is not guaranteed.

Indeed. You reminded me that `async` shouldn't be used. Now I switched to the promise version so it should be working.

**Comment by dklilley** [comment]
Thanks for making that change!

For me, `realpath` is resolving to `/usr/local/MATLAB/R2023b/bin/matlab`, which is expected. However, for this to work, I believe that `binPath` needs to point to `/usr/local/MATLAB/R2023b/bin/`.

**Comment by MoetaYuko** [comment]
Oops, my bad, this revision must be okay...

**Pull request merged**